### ISSUE TITLE
A11y_Model Builder_Add Data_Focus Order : Focus is shifting to the data source radio button directly at down and then with shift tab focus is moving to link at top.

### ISSUE THREAD

**Issue by manaswita-chougule** [dotnet/machinelearning-modelbuilder#2181]
### GitHubTags:
#A11yMAS; #A11yTCS; #A11ySev2; #DesktopApp; #Visual Studio Client; #SH-ModelBuilder-Win32-Jun2022; #Win32; #WCAG2.4.3; #FocusOrder; #Keyboard;

### Environment Details:
Application Name: Model Builder
Windows Version: Windows 11 22H2 OS Builder 25145.1011
Visual Studio: Microsoft Visual Studio Enterprise 2022 (64-bit) - Preview Version 17.3.0 Preview 2.0

### Repro Steps:
1. Install Visual studio and open Visual studio
2. Navigate to C# project name and right click on it
3. Navigate to "Add" button and press enter
4. Navigate to "Machine learning model" and press enter
5. "Add new item" popup will open
6. Navigate to "Add" button and press enter
7. "Select scenario" popup will open and select "Forecasting" scenario
8. "Select training environment" page will open then select "Next step"
9. "Add Data" page will appear.
10. Check whether the focus is shifting to the data source radio button directly at down and then with shift tab focus is moving to link at top.

### Actual Result:
Focus is shifting to the data source radio button directly at down and then with shift tab focus is moving to link at top.
### Note:
Similar issue is observed with "Train Data".

### Expected Result:
Focus should shift to the control at the top first (top link) and then to the bottom controls (Data source radio buttons), focus order should be top to bottom and left to right.

![add data_focus order (1)](https://user-images.githubusercontent.com/91586101/176685785-62eb638b-e59b-4f4e-974e-d24a6f77232f.gif)

**Comment by beccamc** [comment]
@romalpani Can you help me here? I think we set the focus to the first interactable element. Is this correct? Have I understood the bug correctly?

**Comment by romalpani** [comment]
Hi Becca!

According to accessibility guidelines, the focus should go to the first interactable element. 
https://devdivdesignguide.azurewebsites.net/general-guidelines/designing-for-everyone/#focus

In this case, it would be the hyperlink. But I think, we had decided to bring focus directly to the "environment" selection area because that gets customers to achieve their task faster (one less tab key press).

So, I am going to check with the design team if there are other places we do a similar thing, to confirm our approach.

**Comment by romalpani** [comment]
After discussing with the design team, here's what I think we should do to resolve the bug.

1. Bring focus to the first interactable element - in this case, the **hyperlink** (and I believe all other steps of the wizard too).
2. Reasons

- It is a more predictable interaction flow for sighted keyboard and screen reader users.
- Having to move back (shift-tab) to access the hyperlink does seem counter-intuitive. 
- It is easier to build a mental model with a simpler flow, and eventually, customers will tab right past the hyperlink when they don't need it. 

>

**Comment by beccamc** [comment]
Great - will do. Thanks for the help here Rohan!!

**Comment by beccamc** [comment]
@manaswita-chougule  Please verify this on build 16.13.10.2241902 available from the Marketplace https://marketplace.visualstudio.com/items?itemName=MLNET.ModelBuilder2022

**Comment by beccamc** [comment]
@manaswita-chougule What is the ETA on validation of these bugs? Our SLA is next week so I'd like to make sure we have reaction time if there is a regression. Thanks!

**Comment by manaswita-chougule** [comment]
Verified the issue in below environment:
Microsoft Visual Studio Enterprise 2022 (64-bit) - Preview
Version 17.4.0 Preview 1.0

The issue is fixed, hence closing the bug. Please find below attachment for reference.
![2181_fixed](https://user-images.githubusercontent.com/91586101/187138308-488ad822-045a-4e1e-9be1-ed3e7989948e.gif)
